va_explorer/va_data_management/management/commands/run_dhis.py

```diff
@@ -0,0 +1,242 @@
+            storagev= storagev.drop(['ID'], axis=1)
+            storagev.to_csv("OpenVAFiles/recordStorage.csv", index=False)
+
+            # Ensure all variables are formatted as required
+            vadata = pd.read_csv("OpenVAFiles/recordStorage.csv")
+            vadata["age"] = vadata["age"].astype(float)
+            vadata['dod'][vadata.dod.isnull()] = '1900-01-01'
+            vadata['dob'][vadata.dob.isnull()] = '1900-01-01'
+            vadata['id'] = ["A" + str(i) for i in vadata['id']]
+            for i in range(len(vadata["sex"])):
+                vadata["dob"][i] = dateutil.parser.parse(vadata["dob"][i]).date().strftime("%Y-%m-%d")
+                vadata["dod"][i] = dateutil.parser.parse(vadata["dod"][i]).date().strftime("%Y-%m-%d")
+                vadata["cod"][i] = vadata["cod"][i].strip()
+                vadata["metadataCode"][i] =  metadatacode
+            vadata.to_csv("OpenVAFiles/recordStorage.csv", index=False)
+
+            # dhis settings
+            ntDHIS = collections.namedtuple("ntDHIS",
+                                            ["dhisURL",
+                                             "dhisUser",
+                                             "dhisPassword",
+                                             "dhisOrgUnit"]
+                                            )
+            settingsDHIS = ntDHIS(DHIS2_URL, DHIS2_USER, DHIS2_PASS, DHIS2_ORGUNIT)
+
+            CODCodes =  cod_codes_dhis.objects.filter(codsource="WHO").values()
+            queryCODCodes = pd.DataFrame.from_records(CODCodes)
+            queryCODCodes = queryCODCodes[{"codname", "codcode"}]
+
+            dhisCODCodes = dict(zip(queryCODCodes.codname, queryCODCodes.codcode))
+
+            dhisCODCodes = {}
+            argsDHIS = [settingsDHIS, dhisCODCodes]
+
+            # execute pipeline
+            pipelineDHIS = dhis.DHIS(argsDHIS,"")
+
+            apiDHIS = pipelineDHIS.connect()
+            self.clearFolder("DHIS/blobs/")
+            postLog = pipelineDHIS.postVA(apiDHIS)
+            self.clearFolder("DHIS/blobs/")
+            if postLog['response']['status']=='SUCCESS' and postLog['response']['imported']==len(va_not_in_dhis):
+                numPushed = postLog['response']['imported']
+                numTotal = postLog['response']['total']
+                status = postLog['response']['status']
+            self.stdout.write(f' Uploaded {numPushed} out of {numTotal} verbal autopsies ')
+
+        else:
+            numPushed=0
+            numTotal=0
+            status="Nothing to Post"
+
+        self.syncDHISStatus()
+
+        return numPushed,numTotal,status
+
+    def syncDHISStatus(self):
+        env = environ.Env()
+
+        # DHIS2 VARIABLES
+        DHIS2_USER = env("DHIS2_USER")
+        DHIS2_PASS = env("DHIS2_PASS")
+
+        auth = (DHIS2_USER, DHIS2_PASS)
+        va_in_dhis2 = self.getPushedVA('sv91bCroFFx', auth)
+        va_in_dhis2 = [str(i) for i in va_in_dhis2]
+
+        #print(va_in_dhis2)
```

Comment:
Might consider turning this into a log message at the debug level or removing.

Reply:
removed